packages/database/src/testUtilities/buildAndInsertSurveyResponses.js

```diff
@@ -67,9 +67,11 @@ const buildAndInsertSurveyResponse = async (
  */
 export const buildAndInsertSurveyResponses = async (models, surveyResponses) => {
   const user = await upsertDummyRecord(models.user);
-  return Promise.all(
-    surveyResponses.map(async surveyResponse =>
-      buildAndInsertSurveyResponse(models, user, surveyResponse),
-    ),
-  );
+  const builtResponses = [];
+  for (let i = 0; i < surveyResponses.length; i++) {
+    const surveyResponse = surveyResponses[i];
+    const builtResponse = await buildAndInsertSurveyResponse(models, user, surveyResponse);
+    builtResponses.push(builtResponse);
+  }
+  return builtResponses;
```

Comment:
This caused what must be a rare bug on my machine: 
`error: duplicate key value violates unique constraint "entity_code_key"`

I _think_ it's due to `Promise.all` executing operations in non-deterministic order, which might've lead to a situation where two `findOrCreate()` calls were made at the same time. I guess the real underlying issue here is that `DatabaseModel.findOrCreate()` is non-atomic... but that seems like a bit too big of an issue to tackle as part of this PR...

Reply:
Ah that's no good! Can you create a card for it?

Reply:
Good idea! Have created and added to refinement: https://linear.app/bes/issue/RN-536/rare-bug-where-creating-entities-fail-when-many-attempts-are-made-at